src/details/ArborX_DetailsSortUtils.hpp

```diff
@@ -138,6 +144,40 @@ Kokkos::View<SizeType *, typename ViewType::device_type> sortObjects(
 }
 #endif
 
+#if defined(KOKKOS_ENABLE_SYCL)
+// NOTE returns the permutation indices **and** sorts the input view
+template <typename ViewType, class SizeType = unsigned int>
+Kokkos::View<SizeType *, typename ViewType::device_type>
+sortObjects(Kokkos::Experimental::SYCL const &space, ViewType &view)
+{
+  int const n = view.extent(0);
+
+  using ValueType = typename ViewType::value_type;
+  static_assert(std::is_same<std::decay_t<decltype(space)>,
```

Comment:
This is not the right check. You want `is_accessible_from<typename ViewType::memory space, Kokkos::Experimental::SYCL>::value`

Reply:
We have the same check for `Cuda` and `HIP`.

Reply:
But fixed.